sync/src/relayer/block_proposal_process.rs

```diff
@@ -16,12 +18,33 @@ pub enum Status {
 }
 
 impl<'a> BlockProposalProcess<'a> {
-    pub fn new(message: packed::BlockProposalReader<'a>, relayer: &'a Relayer) -> Self {
-        BlockProposalProcess { message, relayer }
+    pub fn new(
+        message: packed::BlockProposalReader<'a>,
+        relayer: &'a Relayer,
+        peer: PeerIndex,
+    ) -> Self {
+        BlockProposalProcess {
+            message,
+            relayer,
+            peer,
+        }
     }
 
     pub fn execute(self) -> Result<Status, FailureError> {
         let snapshot = self.relayer.shared().snapshot();
+        {
+            let block_proposals = self.message;
+            if (block_proposals.transactions().len() as u64)
+                > snapshot.consensus().max_block_proposals_limit()
```

Comment:
Should be 2 * snapshot.consensus().max_block_proposals_limit();

we send request_proposal_txs in 2 place.
One is `CompactBlockProcess`, we request_proposal_txs for one block,
Another one  is `BlockTransactionsProcess`, we request_proposal_txs for uncles.
we might request two uncles's proposal (the worst case, all fresh).

check https://github.com/nervosnetwork/ckb/blob/develop/sync/src/relayer/block_transactions_process.rs#L114-L124

Reply:
Thanks for pointing out, I never noticed this issue before and think that we should increase `MAX_FRAME_LENGTH_RELAY` to 4MB for extreme cases:

https://github.com/nervosnetwork/ckb/blob/c9ab6bdaaca237913869b66fd85ccb64ef5ab7ed/network/src/lib.rs#L39

Reply:
Updated, please review again, thanks.